optimade/server/routers/utils.py

```diff
@@ -252,21 +251,25 @@ def get_entries(
     include = []
     if getattr(params, "include", False):
         include.extend(params.include.split(","))
-    included = get_included_relationships(results, ENTRY_COLLECTIONS, include)
+
+    included = []
+    if results is not None:
+        included = get_included_relationships(results, ENTRY_COLLECTIONS, include)
 
     if more_data_available:
         # Deduce the `next` link from the current request
         query = urllib.parse.parse_qs(request.url.query)
-        query["page_offset"] = int(query.get("page_offset", [0])[0]) + len(results)
+        if isinstance(results, list):
+            query["page_offset"] = int(query.get("page_offset", [0])[0]) + len(results)  # type: ignore[assignment,list-item]
         urlencoded = urllib.parse.urlencode(query, doseq=True)
         base_url = get_base_url(request.url)
 
         links = ToplevelLinks(next=f"{base_url}{request.url.path}?{urlencoded}")
     else:
         links = ToplevelLinks(next=None)
 
-    if fields or include_fields:
-        results = handle_response_fields(results, fields, include_fields)
+    if results is not None and (fields or include_fields):
+        results = handle_response_fields(results, fields, include_fields)  # type: ignore[assignment]
```

Comment:
You need a "type ignore" here because handle_response_fields returns a list of dicts while results previously was a list of entry resources.
Changing the name of `results` on line 272 and  276 to something like "results_dicts" will prevent this error.

Reply:
I'm confused, there is already a `# type: ignore[assignment]` here?

Reply:
Yes, but it can be removed if you rename `results` on lines 272 and 276 to something other than results.

Reply:
I think in a few cases the code is clearer if the ignores are kept in, rather than making additional temporary variables, so I've left this one in.